server/v2/stf/stf.go

```diff
@@ -401,11 +406,12 @@ func (s STF[T]) validatorUpdates(
 	return ctx.events, valSetUpdates, nil
 }
 
-const headerInfoPrefix = 0x0
+// TODO reserved in consensus OR need a keeper here
+const headerInfoPrefix = 0x37
 
 // setHeaderInfo sets the header info in the state to be used by queries in the future.
 func (s STF[T]) setHeaderInfo(state store.WriterMap, headerInfo header.Info) error {
-	runtimeStore, err := state.GetWriter(appmanager.RuntimeIdentity)
```

Comment:
TODO: should runtime have a store key? if not a consensus keeper is needed here to abstract storage.

Reply:
imo yes, because the consensus module is meant to be consensus engine dependent iirc

Reply:
https://github.com/cosmos/cosmos-sdk/issues/20510